solidity/contracts/isms/ccip-read/AbstractCcipReadIsm.sol

```diff
@@ -0,0 +1,141 @@
+
+    // solhint-disable-next-line const-name-snakecase
+    uint8 public constant moduleType =
+        uint8(IInterchainSecurityModule.Types.CCIP_READ);
+
+    // ============ Mutable Storage ============
+
+    string[] public offchainUrls;
+    bytes public offchainExtraData;
+    bytes public offchainCallData;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when the offchain URLs are updated
+     * @param urls the new URLs
+     */
+    event OffchainUrlsUpdated(string[] urls);
+
+    /**
+     * @notice Emitted when the offchain extraData is updated
+     * @param extraData the new extraData
+     */
+    event OffchainExtraDataUpdated(bytes extraData);
+
+    /**
+     * @notice Emitted when the offchain callData is updated
+     * @param callData the new callData
+     */
+    event OffchainCallDataUpdated(bytes callData);
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Reverts with the data needed to query information offchain
+     * and be submitted via verifyWithProof
+     * @dev See https://eips.ethereum.org/EIPS/eip-3668 for more information
+     * @return bool Ignored
+     */
+    function getOffchainVerifyInfo(bytes calldata)
+        external
+        view
+        returns (bool)
+    {
+        revert OffchainLookup(
+            address(this),
+            offchainUrls,
+            offchainCallData,
+            AbstractCcipReadIsm.verifyWithProof.selector,
+            offchainExtraData
+        );
+        return true;
+    }
+
+    /**
+     * @notice Function to be called with the result of the offchain read
+     * @param response the offchain result
+     * @return bool
+     */
+    function verifyWithProof(bytes calldata response, bytes calldata)
+        external
+        view
+        returns (bool)
+    {
+        // TODO: magic numbers
+        uint256 metadataOffset = uint256(bytes32(response[0:32]));
+        uint256 messageOffset = uint256(bytes32(response[33:64]));
+
+        return
+            verify(
+                response[metadataOffset:messageOffset],
+                response[messageOffset:]
+            );
+    }
+
+    // ============ Internal Functions ============
+
+    /**
+     * @notice Returns the digest to be used for signature verification.
+     * @param _message Formatted Hyperlane message (see Message.sol).
+     * @return digest The digest to be signed by validators
+     */
+    function digest(bytes calldata, bytes calldata _message)
+        internal
+        pure
+        override
+        returns (bytes32)
+    {
+        return keccak256(_message.body());
+    }
+
+    /**
+     * @notice Returns the signature at a given index from the metadata.
+     * @param _metadata ABI encoded module metadata
+     * @param _index The index of the signature to return
+     * @return signature Packed encoding of signature (65 bytes)
+     */
+    function signatureAt(bytes calldata _metadata, uint256 _index)
```

Comment:
ideally the abstract CCIP ism doesnt know about signatures or digests
that being said I understand you wanted to just get something working